nixos/modules/services/networking/peroxide.nix

```diff
@@ -0,0 +1,180 @@
+          <xref linked="opt-services.peroxide.stateDir"/>
+          for the peroxide service to start.
+        '';
+      };
+
+      organization = mkOption {
+        type = types.str;
+        default = "N/A";
+        description = ''
+          Value for the 'Organization' field of the self-signed X.509 certificate.
+        '';
+      };
+
+      hostname = mkOption {
+        type = types.str;
+        default = config.networking.hostName;
+        description = ''
+          Value for the 'CN' field of the self-signed X.509 certificate.
+        '';
+      };
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "peroxide";
+      description = "User as which peroxide is run.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "peroxide";
+      description = "Group as which peroxide is run.";
+    };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+
+        options = {
+          CacheDir = mkOption {
+            type = types.path;
+            default = "${cfg.stateDir}/cache";
+            # See comment on module assertions for an explanation of this
+            # restriction.
+            description = ''
+              The cache directory for peroxide.
+
+              <warning>
+                This <emphasis>must</emphasis> be a subdirectory of
+                <xref linked="opt-services.peroxide.stateDir"/>
+              </warning>
+            '';
+          };
+
+          UserPortImap = mkOption {
+            type = types.port;
+            default = 1143;
+            description = "The port on which to listen for IMAP connections.";
+          };
+
+          UserPortSmtp = mkOption {
+            type = types.port;
+            default = 1025;
+            description = "The port on which to listen for SMTP connections.";
+          };
+
+          ServerAddress = mkOption {
+            type = types.str;
+            default = "[::0]";
+            example = "localhost";
+            description = "The address on which to listen for connections.";
+          };
+        };
+      };
+      default = { };
+      description = ''
+        Configuration for peroxide.  See
+        <link xlink:href="https://github.com/ljanyst/peroxide/blob/v${pkgs.peroxide.version}/config.example.yaml"/>
+        for an example configuration.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.tmpfiles.rules = [
+      "d '${cfg.stateDir}' 0750 ${cfg.user} ${cfg.group} - -"
+
+      "L+ /etc/peroxide - - - - ${cfg.stateDir}"
+    ];
+
+    systemd.services.peroxide = {
+      requires = [ "network.target" ];
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart = "${pkgs.peroxide}/bin/peroxide -log-file=${cfg.stateDir}/peroxide.log";
```

Comment:
The package should be configurable and rather be in path.

Reply:
OK, I'll make it a module option, but I'm not sure what you mean by "and rather be in path."